pkgs/controllers/certificaterequest.go

```diff
@@ -135,6 +139,15 @@ func (r *CertificateRequestController) Reconcile(ctx context.Context, cr *certma
 	}
 
 	pem, err := p.Sign(ctx, cr)
+
+	var apiError cfapi.APIError
+	if errors.As(err, &apiError) {
+		if apiError.Code == originDBWriteErrorCode {
+			log.Error(err, "requeue-ing after API error")
+			return reconcile.Result{Requeue: true}, nil
```

Comment:
Should we delay the requeue as to not overwhelm the backend if it’s possibly already having a bad time?

Reply:
The controller requeues with `workqueue.AddRateLimited`, which is an exponential backoff (and is subject to a workqueue qps limit as well). Providing our own RequeueAfter disables this, and I'm not convinced we'd pick a better value, especially in lieu of failure counts.

Reply:
Just curious, why this form rather than `return reconcile.Result{}, err`?

They're functionally equivalent as far as the backoff, but returning the error has controller-runtime increment its `controller_runtime_reconcile_errors_total` metric, which seems appropriate for this case.